### PR TITLE
Fix migration bug

### LINKED ISSUE #4328: Migration failed on disk balancer
### Steps to reproduce the behavior (Required)
The migration failed when balancing tablet.
```
W0322 15:50:38.332617 15226 engine_storage_migration_task.cpp:148] tablet_meta already exist. tablet:17795.1429620549.b042074d096e3a8d-591818ba87e51bbf
W0322 15:50:38.332703 15226 task_worker_pool.cpp:1094] storage migrate failed. status:Already exist: tablet_meta already exist. tablet: 17795.1429620549.b042074d096e3a8d-591818ba87e51bbf
/home/disk1/chaoyli/starrocks/be/src/storage/storage_engine.cpp:980 task->execute(), signature:17795
W0322 15:53:58.367429 15225 engine_storage_migration_task.cpp:148] tablet_meta already exist. tablet:17795.1429620549.b042074d096e3a8d-591818ba87e51bbf
W0322 15:53:58.367473 15225 task_worker_pool.cpp:1094] storage migrate failed. status:Already exist: tablet_meta already exist. tablet: 17795.1429620549.b042074d096e3a8d-591818ba87e51bbf
/home/disk1/chaoyli/starrocks/be/src/storage/storage_engine.cpp:980 task->execute(), signature:17795
W0322 15:57:18.410866 15223 engine_storage_migration_task.cpp:148] tablet_meta already exist. tablet:17795.1429620549.b042074d096e3a8d-591818ba87e51bbf
W0322 15:57:18.410904 15223 task_worker_pool.cpp:1094] storage migrate failed. status:Already exist: tablet_meta already exist. tablet: 17795.1429620549.b042074d096e3a8d-591818ba87e51bbf
/home/disk1/chaoyli/starrocks/be/src/storage/storage_engine.cpp:980 task->execute(), signature:17795
```

### StarRocks version (Required)
 - branch-2.2 f0805b8e084ba625348c5c4d739a8ad4111e13aa

### PR DESCRIPTION
**Pull request by ABingHuang**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #4328

## Problem Summary(Required) ：

When clone tablet from path A to path B, and then back to path A, it will failed because the original tablet remains in path A. The original tablet is GCed async every one hour. Fix it by removing shutdown tablet when clone. And add ut to test this case.

### CODE REVIEW

**Comment by ABingHuang** [comment]
run starrocks_fe_unittest

File: be/src/storage/task/engine_storage_migration_task.cpp [hunk 1]
```diff
@@ -138,15 +164,20 @@ Status EngineStorageMigrationTask::_storage_migrate(TabletSharedPtr tablet) {
             return Status::AlreadyExist(fmt::format("Path already exist. schema_hash_path: {}", schema_hash_path));
         }
 
+        // double check
```
Comment (chaoyli) [inline be/src/storage/task/engine_storage_migration_task.cpp:167]: Remove the code paths annotated.

File: be/src/storage/task/engine_storage_migration_task.cpp [hunk 2]
```diff
@@ -127,9 +126,36 @@ Status EngineStorageMigrationTask::_storage_migrate(TabletSharedPtr tablet) {
...
         }
 
+        TabletMetaSharedPtr stale_tablet_meta(new (std::nothrow) TabletMeta());
+        Status st = TabletMetaManager::get_tablet_meta(_dest_store, _tablet_id, _schema_hash, stale_tablet_meta.get());
+        if (st.ok() && stale_tablet_meta->tablet_state() == TABLET_SHUTDOWN) {
+            // When tablet state is TABLET_SHUTDOWN, it means it's a tablet to be GCed
+            // Try to delete it right now.
+            // If delete success, go on the process, or return failure.
+            // If there is queries running on the tablet, should return failure
+            Status st = StorageEngine::instance()->tablet_manager()->delete_shutdown_tablet(_tablet_id);
```
Comment (chaoyli) [inline be/src/storage/task/engine_storage_migration_task.cpp:136]: delete and remove should be unified into one word.

Comment (chaoyli) [inline be/src/storage/task/engine_storage_migration_task.cpp:136]: delete and remove should be unified into one word.

**Review by chaoyli — CHANGES_REQUESTED** [review]

**Review by chaoyli — APPROVED** [review]

**Review by decster — APPROVED** [review]

**Pull request merged**